lib/factory.rb

```diff
@@ -1,30 +1,109 @@
+# frozen_string_literal: true
+
+# Struct clone
 class Factory
-  attr_accessor :class_name, :class_attributes
-  def initialize (class_name, *class_attributes)
-    @class_name = class_name
-    @class_attributes = class_attributes
-  end
+  def self.new(*class_attributes, &block)
+    if class_attributes.first.is_a?(String)
+      if class_attributes.first =~ /^[A-Z]\w+(([A-Z]\w+)?)+/
```

Comment:
https://stackoverflow.com/questions/577653/difference-between-a-z-and-in-ruby-regular-expressions

Reply:
Class name can contain `_`?

Reply:
Your regexp is also insufficient, `^[A-Z]\w+` equals to your one.